File: Leetcode/0001 Two Sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        // HashSet<int> seen = new HashSet<int>();
        Dictionary<int,int> seen = new Dictionary<int,int>();
        
        for(int i = 0;i < nums.Length;i++){
            int needed = target - nums[i];
            if(seen.ContainsKey(needed)){
                return new int[]{i,seen[needed]};
            }
                
            else{
                seen[nums[i]] = i;
            }
        }
        return new int[]{};
    }
}

File: Leetcode/0001 Two Sum.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        Map<Integer, Integer> seenAtIndexByNumber = new HashMap<>();
        for (int i = 0; i < numbers.length; ++i) {
            int neededNumber = target - numbers[i];
            if(seenAtIndexByNumber.containsKey(neededNumber)) {
                return new int[]{seenAtIndexByNumber.get(neededNumber), i};
            }
            seenAtIndexByNumber.put(numbers[i], i);
        }
        return new int[]{};   
    }
}
